### ISSUE TITLE
what's “curvature” in e2e module data set?

### ISSUE THREAD

**Issue by abinlee** [ApolloAuto/apollo#658]
Could you explain more details about how do you calculate/record “curvature “ fields in data set and how to use steering model predict value, if predicted result is "curvature", for real-time steering control?

**Comment by yanys1** [comment]
The map survey will capture rtk-GPS traces data, then we can calculate and smooth the curvature data. There are two reasons for this: curvature is more general, not affected by the vehicle's own parameters, such as steering ratio, wheel base and so on. The relationship between curvature and steering wheel is ralatively simply, in low speed through it can be reversed to steering wheel by   Ackermannn model, in high speed it also can be fitted by a simple network.

**Comment by abinlee** [comment]
Can we say E2E model need GPS data for real-time inference (not only camera)?

**Comment by yanys1** [comment]
During training model, we need the camera data as  the input, curvature as the output. We only need the camera when real-time predict.

**Comment by abinlee** [comment]
After I trained model and used test data to check steering prediction, the whole curvatures were near 3.14xxxxxx.
Is this right prediction?

**Comment by yanys1** [comment]
The test set include attr data(ground truth) as a reference, and the format of the test set is consistent with that of the training set, you can use it.

**Comment by xuyangit** [comment]
some questions for me：
1. is learning curvature harder than steer angle？When I test the e2e module，the loss is always in the same order to curvature itself
2. there are six curvatures given in the data，the e2e code just uses one of them，why？
3. To retrieve the steer angle, it seems the wheel base and car length should be given, currently there seems to be no way to find these information, hope you can help me with this.
Really tks!

**Comment by HouYuu** [comment]
To calculate the steer angle, it seems the information of data acquisition vehicle should be given, please help me with this, thank you. @yanys1

**Comment by linpanusst** [comment]
@abinlee 
Can you tell me that if apollo support e2e training algorithms?
If I want to run a demo which is belong to e2e modules in apollo, what should I do or prepare?
And how to run this demo without a car and px2?
Can I just simulate in the computers?

**Comment by natashadsouza** [comment]
Please let me know if you still need support on your request. Feel free to re-open the issue.

**Comment by huangchaoxing** [comment]
Why the dataset has 6 curvatues in each time step
